python/SystemML.py

```diff
@@ -0,0 +1,154 @@
+
+    Parameters
+    ----------
+    results: JavaObject
+        A Java MLResults object as returned by calling ml.execute()
+
+    sc: SparkContext
+        SparkContext
+    """
+    def __init__(self, results, sc):
+        self._java_results = results
+        self.sc = sc
+
+    def __repr__(self):
+        return "MLResults"
+
+    def get(self, *outputs):
+        """
+        Parameters
+        ----------
+        outputs: string, list of strings
+            Output variables as defined inside the DML script.
+        """
+        outs = [_java2py(self.sc, self._java_results.get(out)) for out in outputs]
+        if len(outs) == 1:
+            return outs[0]
+        return outs
+
+
+class Dml(object):
+    """
+    DML instance.
+
+    Parameters
+    ----------
+    path: string
+        Can be either a file path to a DML script or a DML script itself.
+    """
+    def __init__(self, path):
+        self.path = path
+        self._input = {}
+        self._output = []
+
+    def input(self, *args, **kwargs):
+        """
+        Parameters
+        ----------
+        args: name, value tuple
+            where name is a string and currently supported value formats
+            are double, string, rdds and list of such object.
+
+        kwargs: dict of name, value pairs
+            To know what formats are supported for name and value, look above.
+        """
+        if args and len(args) != 2:
+            raise ValueError("Expected name, value pair.")
+        elif args:
+            self._input[args[0]] = args[1]
+        for name, value in kwargs.items():
+            self._input[name] = value
+        return self
+
+    def out(self, *names):
+        """
+        Parameters
+        ----------
+        outputs: string, list of strings
+            Output variables as defined inside the DML script.
+        """
+        self._output.extend(names)
+        return self
+
+
+class MLContext(object):
+    """
+    Wrapper around the new SystemML MLContext.
+
+    Parameters
+    ----------
+    sc: SparkContext
+        SparkContext
+    """
+    def __init__(self, sc):
+        if not isinstance(sc, SparkContext):
+            raise ValueError("Expected sc to be a SparkContext, got " % sc)
+        self._sc = sc
+        self._ml = sc._jvm.org.apache.sysml.api.mlcontext.MLContext(sc._jsc)
+
+    def __repr__(self):
+        return "MLContext"
+
+    def execute(self, dml):
+        """
+        Execute a DML script.
+
+        Parameters
+        ----------
+        dml: DML instance
+            DML instance defined with the appropriate input and output variables.
+
+        Returns
+        -------
+        ml_results: MLResults
+            MLResults instance.
+        """
+        if not isinstance(dml, Dml):
+            raise ValueError("Expected dml to be an instance of Dml")
+        path = dml.path
+        if isinstance(path, str):
+            if os.path.exists(path):
```

Comment:
You might want to add some error-handling code here in case the user mistypes a path. The code could, for example, use a regex to match strings that can only be file paths (because they won't parse as DML), and print "path ${path} not found" if the path doesn't exist.